pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -577,14 +609,37 @@ lib.extendDerivation
        disallowedReferences = [ ];
        disallowedRequisites = [ ];
      });
-
-     inherit meta passthru overrideAttrs;
-   } //
+     inherit passthru;
+   })
+   // {
+     inherit meta overrideAttrs;
+   }
    # Pass through extra attributes that are not inputs, but
    # should be made available to Nix expressions using the
    # derivation (e.g., in assertions).
-   passthru)
-  (derivation (derivationArg // lib.optionalAttrs envIsExportable checkedEnv));
+   // passthru)
+  (
+      (
+        if __cleanAttrs == true
+        then builtins.intersectAttrs { version = null; } drvAttrs
```

Comment:
Why is version propagated? I guess it's somewhat important and stable enough. Should definitely be a comment here